### PR TITLE
Full reset password flow in UI

### PR DESCRIPTION
**Pull request by danielkryska**
- Send reset password url to email
- Reset password

### CODE REVIEW

**Review by jswk — CHANGES_REQUESTED** [review]
The `POST token-create` requires the email passed as a request param, and not as a request body, i.e.
```
POST /token-create?email=foo@bar.pl
```
and not
```
POST /token-create

{"email":"foo@bar.pl"}
```

**Comment by danielkryska** [comment]
Still backend errors occurs

```
ERROR o.s.a.i.SimpleAsyncUncaughtExceptionHandler - Unexpected exception occurred invoking async method: public void pl.cyfronet.s4e.listener.PasswordListener.handle(pl.cyfronet.s4e.event.OnPasswordResetTokenEmailEvent) throws pl.cyfronet.s4e.ex.NotFoundException
s4e-backend_1  | java.lang.IllegalArgumentException: From address must not be null
s4e-backend_1  | 	at org.springframework.util.Assert.notNull(Assert.java:201)
s4e-backend_1  | 	at org.springframework.mail.javamail.MimeMessageHelper.setFrom(MimeMessageHelper.java:571)
s4e-backend_1  | 	at pl.cyfronet.s4e.service.MailService.sendEmail(MailService.java:58)
s4e-backend_1  | 	at pl.cyfronet.s4e.service.MailService.sendEmail(MailService.java:47)
s4e-backend_1  | 	at pl.cyfronet.s4e.listener.PasswordListener.handle(PasswordListener.java:59)
s4e-backend_1  | 	at pl.cyfronet.s4e.listener.PasswordListener$$FastClassBySpringCGLIB$$273883d3.invoke(<generated>)
s4e-backend_1  | 	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
s4e-backend_1  | 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771)
s4e-backend_1  | 	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
s4e-backend_1  | 	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749)
s4e-backend_1  | 	at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115)
s4e-backend_1  | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
s4e-backend_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
s4e-backend_1  | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
s4e-backend_1  | 	at java.base/java.lang.Thread.run(Thread.java:834)

**Comment by jswk** [comment]
The problem with sending mails in development has been fixed in #1072.

**Review by jswk — COMMENTED** [review]
I'd say that for a "full flow" you'd need error cases handled too.

For example, if you generate at least two tokens for an account, and enter the earlier token, then you get a generic 404 handler, which seems to be misleading.
image

In that place you can also get 401 if the token has expired.

Moreover, there is an endpoint for token validation too (`GET /token-validate`), which returns the same codes prior to user entering a password, so you can provide user with information about token expiry or non-existence prior to her entering a password.

Nevertheless, the happy-path works. So it's acceptable for an initial iteration, just not as a full solution.

(A suggestion about copyright updates too, my IntelliJ only updates the year in files I change when I choose to "Update copyright" upon commit. The changes would be much easier to track down if you had it configured that way too. It's strange that it doesn't work this way out-of-the-box actually.)

File: s4e-web/src/app/state/session/session.service.ts [hunk 1]
```diff
@@ -83,6 +93,53 @@ export class SessionService {
...
+    };
+    const url = `${environment.apiPrefixV1}/password-reset`;
+    return this._http.post(url, {token, password}, headers)
+      .pipe(
+        catchError((error: HttpErrorResponse) => {
+          switch (error.status) {
+            case HTTP_404_NOT_FOUND:
+              this._notificationService.addGeneral({
+                type: 'error',
+                content: 'Token dla resetu hasła niestnieje'
```
Comment (jswk) [inline s4e-web/src/app/state/session/session.service.ts:124]: Typo: niestnieje -> nie istnieje

**Review by jswk — APPROVED** [review]
Apart from the typo the flow now includes the error cases handling and is fine to merge.

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**